httpcache.go

```diff
@@ -184,6 +184,13 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 			cachedResp.StatusCode = http.StatusOK
 
 			resp = cachedResp
+		} else if (err != nil || (cachedResp != nil && cachedResp.StatusCode >= 500)) &&
```

Comment:
It should be resp's statusCode that is being checked. A test for this would be good too.